logging/google/cloud/logging/handlers/app_engine.py

```diff
@@ -14,60 +14,55 @@
+from google.cloud.logging.handlers.transports import BackgroundThreadTransport
+from google.cloud.logging.resource import Resource
 
-_LOG_PATH_TEMPLATE = '/var/log/app_engine/app.{pid}.json'
-_MAX_LOG_BYTES = 128 * 1024 * 1024
-_LOG_FILE_COUNT = 3
+_GAE_PROJECT_ENV = 'GCLOUD_PROJECT'
+_GAE_SERVICE_ENV = 'GAE_SERVICE'
+_GAE_VERSION_ENV = 'GAE_VERSION'
 
 
-class AppEngineHandler(logging.handlers.RotatingFileHandler):
-    """A handler that writes to the App Engine fluentd Stackdriver log file.
+class AppEngineHandler(CloudLoggingHandler):
+    """A handler that directly makes Stackdriver logging API calls.
 
-    Writes to the file that the fluentd agent on App Engine Flexible is
-    configured to discover logs and send them to  Stackdriver Logging.
-    Log entries are wrapped in JSON and with appropriate metadata. The
-    process of converting the user's formatted logs into a JSON payload for
-    Stackdriver Logging consumption is implemented as part of the handler
-    itself, and not as a formatting step, so as not to interfere with
-    user-defined logging formats.
-    """
-
-    def __init__(self):
-        """Construct the handler
+    This handler can be used to route Python standard logging messages directly
+    to the Stackdriver Logging API.
 
-        Large log entries will get mangled if multiple workers write to the
-        same file simultaneously, so we'll use the worker's PID to pick a log
-        filename.
-        """
-        self.filename = _LOG_PATH_TEMPLATE.format(pid=os.getpid())
-        super(AppEngineHandler, self).__init__(self.filename,
-                                               maxBytes=_MAX_LOG_BYTES,
-                                               backupCount=_LOG_FILE_COUNT)
+    This handler supports both an asynchronous and synchronous transport.
 
-    def format(self, record):
-        """Format the specified record into the expected JSON structure.
+    :type client: :class:`google.cloud.logging.client`
+    :param client: the authenticated Google Cloud Logging client for this
+                   handler to use
 
-        :type record: :class:`~logging.LogRecord`
-        :param record: the log record
+    :type transport: type
+    :param transport: Class for creating new transport objects. It should
+                      extend from the base :class:`.Transport` type and
+                      implement :meth`.Transport.send`. Defaults to
+                      :class:`.BackgroundThreadTransport`. The other
+                      option is :class:`.SyncTransport`.
+    """
 
-        :rtype: str
-        :returns: JSON str to be written to the log file
-        """
-        message = super(AppEngineHandler, self).format(record)
-        return format_stackdriver_json(record, message)
+    DEFAULT_LOGGER_NAME = 'app'
+
+    def __init__(self, client,
+                 transport=BackgroundThreadTransport):
+        super(AppEngineHandler, self).__init__(client, name=self.DEFAULT_LOGGER_NAME,
```

Comment:
nit: please avoid hanging indents (they make me sad), prefer instead to line break at the earliest possible opening parenthesis:

```
super(AppEngineHandler, self).__init__(
    client,
    name=_DEFAULT_LOGGER_NAME,
    transport=transport,
    resource=self.gae_resource)
```